bundles/org.openhab.binding.ism8/src/main/java/org/openhab/binding/ism8/internal/Ism8Handler.java

```diff
@@ -0,0 +1,170 @@
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.ism8.internal;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.library.types.QuantityType;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.openhab.binding.ism8.server.DataPointChangedEvent;
+import org.openhab.binding.ism8.server.IDataPoint;
+import org.openhab.binding.ism8.server.IDataPointChangeListener;
+import org.openhab.binding.ism8.server.Server;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link Ism8Handler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Hans-Reiner Hoffmann - Initial contribution
+ */
+@NonNullByDefault
+public class Ism8Handler extends BaseThingHandler implements IDataPointChangeListener {
+    private final Logger logger = LoggerFactory.getLogger(Ism8Handler.class);
+
+    private @Nullable Ism8Configuration config;
+    private @Nullable Server server;
+
+    public Ism8Handler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.debug("Ism8: Handle command = {} {}", channelUID.getId(), command);
+        if (isLinked(channelUID)) {
+            Channel channel = null;
+            for (Channel ch : getThing().getChannels()) {
+                if (ch.getUID().getId().equals(channelUID.getId())) {
+                    channel = ch;
+                    break;
+                }
+            }
+
+            if (channel != null && this.server != null) {
+                Server svr = this.server;
+                if (channel.getConfiguration().containsKey("id") && channel.getConfiguration().containsKey("write")
+                        && channel.getConfiguration().get("write").toString().equalsIgnoreCase("true")) {
+                    int id = Integer.parseInt(channel.getConfiguration().get("id").toString());
+                    logger.info("Channel '{}' writting into ID '{}'", channel.getUID().getId(), id);
+                    updateState(channelUID, new QuantityType<>(command.toString()));
+
+                    IDataPoint dataPoint = null;
+                    for (IDataPoint dp : svr.getDataPoints()) {
+                        if (dp.getId() == id) {
+                            dataPoint = dp;
+                            break;
+                        }
+                    }
+
+                    if (dataPoint != null) {
+                        try {
+                            svr.sendData(dataPoint.createWriteData(command));
+                        } catch (Exception e) {
```

Comment:
The server method 'sendData' uses an OutputStream to write the bytes into the stream. I'm not sure what kind of Exceptions does the OutputStream may throw. In any case was the sending the data not successfully and this needs to be logged / handled. I wanted to avoid any unhandled exception that is may throwed upwards.

Reply:
`sendData()` is declared with `throws Exception`, but throws actually only `IOException`s. So, you should change the `throws` declaration of `sendData()` to `throws IOException`. The compiler will force you to handle all *checked* exceptions. So, don't be afraid of missing any *checked* exception. *Unchecked* exceptions like `RuntimeExceptions` are not intended to be catched, normally. Although, openHAB handle those uncatched exception by logging them.

Reply:
I have changed it into IOException